addons/account/models/account_tax.py

```diff
@@ -1615,16 +1638,27 @@ def _round_base_lines_tax_details(self, base_lines, company, tax_lines=None):
             delta_tax_amount_currency = tax_amounts['raw_tax_amount_currency'] - tax_amounts['tax_amount_currency']
             delta_tax_amount = tax_amounts['raw_tax_amount'] - tax_amounts['tax_amount']
 
-            tax_data = next(
-                x
-                for x in tax_details['taxes_data']
+            index, tax_data = next(
+                (i, x)
+                for i, x in enumerate(tax_details['taxes_data'])
                 if x['tax'] == tax and x['is_reverse_charge'] == is_reverse_charge
             )
             tax_amounts['reference_tax_data'] = tax_data
+            if currency.is_zero(delta_tax_amount_currency) and company.currency_id.is_zero(delta_tax_amount):
+                continue
+
             tax_data['tax_amount_currency'] += delta_tax_amount_currency
             tax_data['tax_amount'] += delta_tax_amount
+            tax_amounts['tax_amount_currency'] += delta_tax_amount_currency
+            tax_amounts['tax_amount'] += delta_tax_amount
 
-        # Dispatch the delta of base amounts accross the base lines.
+            if index == 0:
```

Comment:
Why only if it's the first tax detail?

Reply:
If you have 2 taxes on the same base line, both having a separated grouping key, you want to fix only once the base amount/tax amounts. If you don't do that, you could duplicate some delta amounts

Reply:
I understand it with you loop and check/modify all the lines.
But here you only modify one line with the `next` function :thinking:

Reply:
@william-andre The tax details is about a single line and you can't have more than one combination tax/reverse_charge per line